src/plugins/auto/auto_schedule.cpp

```diff
@@ -125,105 +124,180 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
         std::list<DeviceInformation> validDevices =
             _autoSContext->_plugin->GetValidDevice(_autoSContext->_devicePriorities,
                                                    _loadContext[ACTUALDEVICE].networkPrecision);
+        // When the hint is ctput and there is only one device, the single-device logic is used
         if (validDevices.size() == 1) {
-            // When the hint is ctput and there is only one device, the single-device logic is used instead of
-            // the MULTI logic
-            // can not change _autoSContext->_performanceHint to THROUGHPUT, because GetMetric needs to return CTPUT
             _loadContext[ACTUALDEVICE].deviceInfo = validDevices.front();
             _loadContext[ACTUALDEVICE].deviceInfo.config[CONFIG_KEY(PERFORMANCE_HINT)] =
                 IE::PluginConfigParams::THROUGHPUT;
-            isCumulative = false;
-        } else {
-            // When the hint is ctput and there are more than one device, the MULTI logic is used
-            std::string deviceName = "MULTI:";
+        } else if (validDevices.size() > 1) {
+            _loadContext[ACTUALDEVICE].isEnabled = false;
+            _autoSContext->_devicePriorities.clear();
+            std::copy(std::begin(validDevices),
+                      std::end(validDevices),
+                      std::back_inserter(_autoSContext->_devicePriorities));
+            // Total number of devices in CTPUT
+            _nCTputDeviceNums = validDevices.size();
+            DeviceInformation cpuDeviceInformation;
+            const auto CPUIter =
+                std::find_if(validDevices.begin(), validDevices.end(), [](const DeviceInformation& d) -> bool {
+                    return d.deviceName.find("CPU") != std::string::npos;
+                });
+            if (CPUIter != validDevices.end()) {
+                if (CPUIter->config.find(ov::affinity.name()) == CPUIter->config.end()) {
+                    LOG_DEBUG_TAG("set affinity to CORE for CPU");
+                    // no user set affinity then set CORE to CPU
+                    CPUIter->config.insert(
+                        {ov::affinity.name(), ov::affinity(ov::Affinity::CORE).second.as<std::string>()});
+                }
+                cpuDeviceInformation = *CPUIter;
+                validDevices.erase(CPUIter);
+            }
+            // Generate contexts for loading each device
+            _pCTPUTLoadContext.reset(new AutoLoadContext[_nCTputDeviceNums]);
+            int idx = 0;
             for (auto& device : validDevices) {
-                deviceName += device.deviceName;
-                deviceName += ((device.deviceName == validDevices.back().deviceName) ? "" : ",");
+                _pCTPUTLoadContext[idx].deviceInfo = device;
+                _pCTPUTLoadContext[idx].deviceInfo.config[CONFIG_KEY(PERFORMANCE_HINT)] = IE::PluginConfigParams::THROUGHPUT;
+                idx++;
```

Comment:
@wangleis do we need to support latency mode with auto_ctput?

Reply:
@songbell Will not support latency mode in 2023.0 for same behavior between MULTI and AUTO CTPUT. Need customer use case to clarify the requirement first.

Reply:
Need clarification in document for this change in 2023.0

Reply:
What is the behavior if user -d MULTI:GPU,CPU -hint LATENCY?  Will threw exception?   We can add the expected behavior in CVS-104939, and document in  CVS-105015

Reply:
I think this is not a huge change for current code, -d MULTI with latency, we just switch multi->auto ctput in auto plugin(set the flag in autoSContext), and just carry on other configs from set_property or compile_model, and pass to load lamdas in AUTO.
so shall we update the logic to support latency, or update the document (and communicate the changes)? what do you think? @peterchen-intel